packages/client/src/utils.test.ts

```diff
@@ -32,3 +32,54 @@ describe('decodeToken', () => {
     expect(() => decodeToken(JWT)).toThrowError(StructError);
   });
 });
+
+describe('generateCallbackUri', () => {
+  test('generate callback url with redirectUri', () => {
+    const redirectUri = 'http://localhost:3000';
+    expect(generateCallbackUri({ redirectUri })).toEqual('http://localhost:3000');
+  });
+
+  test('generate callback url with redirectUri and code', () => {
+    const redirectUri = 'http://localhost:3000';
```

Comment:
Looks like those constants are shared cross all the test cases. We could move them to the upper scope, so no need to define multiple times.